test/yargs.js

```diff
@@ -2339,4 +2339,36 @@ describe('yargs dsl tests', () => {
         })
     })
   })
+
+  describe('onFinishCommand', () => {
+    it('use with promise', (done) => {
+      const result = 'noop-result'
+      yargs(['noop']).command('noop', 'a noop command', noop, () => {
+        return Promise.resolve(result)
+      }).onFinishCommand(async (commandResult) => {
```

Comment:
Why do you need this function to be `async`?

Reply:
Remove `async`, thx